test/AppContainer/AppContainer.dev.js

```diff
@@ -181,6 +181,211 @@ function runAllTests(useWeakMap) {
 
         expect(wrapper.text()).toBe('new render + old state');
       });
+
+      it('replaces children class methods', () => {
+        const spy = createSpy();
+
+        class App extends Component {
+          componentWillMount() {
+            this.state = 'new';
+          }
+
+          shouldComponentUpdate() {
+            return false;
+          }
+
+          handleClick() {
+            spy('foo');
+          }
+
+          render() {
+            return <span onClick={this.handleClick}>old render</span>;
+          }
+        }
+        RHL.register(App, 'App', 'test.js');
+
+        const wrapper = mount(<AppContainer><App /></AppContainer>);
+        wrapper.find('span').simulate('click');
+        expect(spy).toHaveBeenCalledWith('foo');
+
+        spy.reset();
+        {
+          class App extends Component {
+            componentWillMount() {
+              this.state = 'new';
+            }
+
+            shouldComponentUpdate() {
+              return false;
+            }
+
+            handleClick() {
+              spy('bar');
+            }
+
+            render() {
+              return <span onClick={this.handleClick}>new render</span>;
+            }
+          }
+          RHL.register(App, 'App', 'test.js');
+          wrapper.setProps({ children: <App /> });
+        }
+
+        wrapper.find('span').simulate('click');
+        expect(spy).toHaveBeenCalledWith('bar');
```

Comment:
@calesce it might be worth it to test that the method is replaced without remounting the component, eg. like it is done here https://github.com/gaearon/react-hot-loader/blob/next/test/AppContainer/AppContainer.dev.js#L182